Handle Enumerate in HTMLBodyElement and HTMLFrameElement

stwrt opened the pull request:
#### 948da4088dfbbd561749b29c8f5ee9a0722c2edd
<pre>
Handle Enumerate in HTMLBodyElement and HTMLFrameElement
<a href="https://bugs.webkit.org/show_bug.cgi?id=244178">https://bugs.webkit.org/show_bug.cgi?id=244178</a>

Reviewed by Brent Fulgham.

Add support for enumerating over additional methods (listed below) in HTMLBodyElement and HTMLFrameSetElement.

onblur
onerror
onfocus
onfocusin
onfocusout
onload
onresize
onscroll

* Source/WebCore/html/HTMLBodyElement.idl:
* Source/WebCore/html/HTMLFrameSetElement.idl:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/Body-FrameSet-Event-Handlers-expected.txt
* LayoutTests/imported/w3c/web-platform-tests/dom/events/Body-FrameSet-Event-Handlers-expected.txt

Canonical link: <a href="https://commits.webkit.org/253840@main">https://commits.webkit.org/253840@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a2e732c8b6184dd21a0aff25a105eb5a0c575d08

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87203 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31290 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18031 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96235 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149779 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91179 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29654 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25884 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/79319 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91215 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92819 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/23957 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74008 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/79319 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/78932 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79178 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/79319 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27379 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13015 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27326 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14030 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2702 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29011 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/36886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28950 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33307 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/html/HTMLBodyElement.idl:
```diff
@@ -29,15 +29,14 @@
     [CEReactions=NotNeeded, Reflect] attribute [LegacyNullToEmptyString] DOMString text;
     [CEReactions=NotNeeded, Reflect] attribute [LegacyNullToEmptyString] DOMString vLink;
 
-    // Non-standard
```
We probably want to keep this comment for the `onwebkit*` handlers below. And perhaps for focusin/out we could refer to the GitHub issue, though it's a somewhat widely known problem so maybe that's okay.

brentfulgham reviewed (approved):
LGTM! Thank you for revising things to identify the non-standard bits.

stwrt committed:
Handle Enumerate in HTMLBodyElement and HTMLFrameElement
https://bugs.webkit.org/show_bug.cgi?id=244178

Reviewed by Brent Fulgham.

Add support for enumerating over additional methods (listed below) in HTMLBodyElement and HTMLFrameSetElement.

onblur
onerror
onfocus
onfocusin
onfocusout
onload
onresize
onscroll

* Source/WebCore/html/HTMLBodyElement.idl:
* Source/WebCore/html/HTMLFrameSetElement.idl:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/Body-FrameSet-Event-Handlers-expected.txt
* LayoutTests/imported/w3c/web-platform-tests/dom/events/Body-FrameSet-Event-Handlers-expected.txt

Canonical link: https://commits.webkit.org/253840@main
